src/simsopt/mhd/bootstrap.py

```diff
@@ -0,0 +1,740 @@
+    same ``s`` grid. The bootstrap current
+    :math:`\left<\vec{J}\cdot\vec{B}\right>` will be computed on this
+    same set of flux surfaces.
+
+    Args:
+        s: A 1D array of values of normalized toroidal flux.
+        ne: A :obj:`~simsopt.mhd.profiles.Profile` object with the electron density profile.
+        Te: A :obj:`~simsopt.mhd.profiles.Profile` object with the electron temperature profile.
+        Ti: A :obj:`~simsopt.mhd.profiles.Profile` object with the ion temperature profile.
+        Zeff: A :obj:`~simsopt.mhd.profiles.Profile` object with the profile of the average
+            impurity charge :math:`Z_{eff}`. Or, a single number can be provided if this profile is constant.
+            Or, if ``None``, Zeff = 1 will be used.
+        G: A 1D array with the flux function multiplying :math:`\nabla\varphi` in the Boozer covariant representation,
+            equivalent to :math:`R B_{toroidal}` in axisymmetry.
+        R: A 1D array with the effective major radius to use when evaluating
+            the collisionality in the Sauter/Redl formulae.
+        iota: A 1D array with the rotational transform.
+        epsilon: A 1D array with the effective inverse aspect ratio to use for
+            evaluating the collisionality in the Sauter/Redl formulae.
+        f_t: A 1D array with the effective trapped fraction.
+        psi_edge: The toroidal flux (in Webers) divided by (2pi) at the boundary s=1
+        helicity_N: 0 for quasi-axisymmetry, or +/- the number of field periods for quasi-helical symmetry.
+            This quantity is used to apply the quasisymmetry isomorphism to map the collisionality
+            and bootstrap current from the tokamak expressions to quasi-helical symmetry.
+
+    Returns:
+        Tuple containing
+
+        - **jdotB**: A 1D array containing the bootstrap current :math:`\left<\vec{J}\cdot\vec{B}\right>`
+          on the specified flux surfaces.
+        - **details**: An object holding intermediate quantities from the computation
+          (e.g. L31, L32, alpha) as attributes
+    """
+
+    if Zeff is None:
+        Zeff = ProfilePolynomial(1.0)
+    if not isinstance(Zeff, Profile):
+        # Zeff is presumably a number. Convert it to a constant profile.
+        Zeff = ProfilePolynomial([Zeff])
+
+    # Evaluate profiles on the grid:
+    ne_s = ne(s)
+    Te_s = Te(s)
+    Ti_s = Ti(s)
+    Zeff_s = Zeff(s)
+    ni_s = ne_s / Zeff_s
+    pe_s = ne_s * Te_s
+    pi_s = ni_s * Ti_s
+    d_ne_d_s = ne.dfds(s)
+    d_Te_d_s = Te.dfds(s)
+    d_Ti_d_s = Ti.dfds(s)
+
+    # Profiles may go to 0 at s=1, so exclude the last 2 grid points:
+    if np.any(ne_s[:-2] < 1e17):
+        logging.warning('ne is surprisingly low. It should have units 1/meters^3')
+    if np.any(Te_s[:-2] < 50):
+        logging.warning('Te is surprisingly low. It should have units of eV')
+    if np.any(Ti_s[:-2] < 50):
+        logging.warning('Ti is surprisingly low. It should have units of eV')
```

Comment:
Are there any checks on collisionality that could be relevant to include here? Is there a specific range in which the Redl formula is valid?